app/src/client/components/buildPage/PropertySchemaParser.ts

```diff
@@ -108,8 +108,9 @@ export class PropertySchemaParser implements PropertySchemaParserInterface {
     if ('json_schema' in schema) {
       Object.entries(schema.json_schema.properties).forEach(([key, property]: [string, any]) => {
         const isReferenceField = this.checkIfRefField(property);
-        if (isReferenceField) {
-          this.handleReferenceField(key, property, defaultValues);
+        const refTypes = this.getRefTypes(property);
+        if (isReferenceField && refTypes.length > 0) {
```

Comment:
Is it normal situation to have referenceField that has no refTypes?

What I wanted to is to ask, is it normal, or check for length possibly covers mistake somewhere else?

Reply:
Actually no, all the reference fields should have at-least one refTypes. As discussed, I will update the logic in checkIfRefField method to return True only if the field has any reference.